scala/README.md

```diff
@@ -638,13 +683,27 @@ Tips & Tricks
     5.stars // Equivalent to a static method call, no implicit conversion actually takes place
     ```
 
+0. Whenever possible, use `private[this]` instead of `private` and `final val` instead of `val` as they enable the Scala compiler and the JVM to perform additional optimizations: direct field access vs. accessor method, or inlined constant vs. field access, respectively. (If `final val` surprised you, remember that it is not redundant, as in Scala `final` means "cannot be overridden", while in Java it may mean both that as well as "cannot be reassigned").
+
 0. Leverage parallel collections, use `.par` judiciously.
 
+0. `import System.{currentTimeMillis => now}` or `import System.{nanoTime => now}` are very useful to have around.
+
 0. Use the `@tailrec` annotation to ensure the compiler can recognize a recursive method is tail-recursive.
 
 0. You can use the `@scala.beans.BeanProperty` and `@BooleanBeanProperty` annotations to automatically generate JavaBeans style getter and setter methods.
 
-0. Use this syntax to pass a sequence as a parameter to a variable length argument list method:
+0. If a method takes a parameter of type `Seq[T]`, consider putting that parameter last and making it a variable-length argument (_vararg_):
```

Comment:
i think this needs to be qualified a bit. varargs are useful when a function will take multiple manually assigned values. for example: a class that rolls up a bunch of instances of classes defined in your project through a fold will always have its arguments passed in manually and be relatively small (thinking of a controller that folds over `Route`s from akka-http here). a function that takes a sequence of things that are generated at runtime will most likely not benefit from this.